TodoAppWithSpringboot/src/main/java/com/webonise/service/TodoService.java

```
@@ -0,0 +1,64 @@
+package com.webonise.service;
+
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import com.webonise.dao.TodoDao;
+import com.webonise.model.Todo;
+import lombok.extern.slf4j.Slf4j;
+
+@Service
+@Slf4j
+public class TodoService {
+
+	@Autowired
+	private TodoDao todoDao;
+
+	public List<Todo> findAll() throws Exception {
+		try {
+			return (List<Todo>) todoDao.findAll();
+		} catch (NullPointerException ex) {
+			log.info("Todos are not selected");
+			throw new NullPointerException("Todo list is not initialized properly.");
+		} catch (Exception ex) {
+			log.info("Exception occured while selecting all todos from database.");
```

can't swallow exceptions like this!